gateway/actions/auth-action.ts

```diff
@@ -1,23 +1,56 @@
-import { createIdentityToken } from "../pastaporto/identity";
+import {
+  createIdentityToken,
+  createRefreshToken,
+} from "../pastaporto/identity";
 import { IS_DEV } from "../config";
 import { PastaportoAction } from "./entities";
 
-export class AuthAction extends PastaportoAction {
+type Options = {
+  identityOnly?: boolean;
+};
+
+type Return = {
+  identityToken: string;
+  refreshToken?: string;
+};
+
+export class AuthAction extends PastaportoAction<Options, Return> {
   async apply(context: any) {
-    const identityToken = await createIdentityToken(
-      parseInt(this.payload["id"], 10),
-    );
+    const sub = `${this.payload["id"]}`;
+    const identityToken = createIdentityToken(sub);
+    let refreshToken;
+
+    // clear previous set cookies (clear identity cookies)
+    context.setCookies.splice(0, context.setCookies.length);
 
     context.setCookies.push({
       name: "identity",
       value: identityToken,
       options: {
         httpOnly: true,
-        maxAge: 60 * 15,
+        maxAge: 60 * 60 * 24 * 7 * 4 * 3,
```

Comment:
can we use a constant here to make it more readable? :)

Reply:
maybe now it's more clear?